services/statestorage/service.go

```diff
@@ -52,7 +56,10 @@ func (s *service) CommitStateDiff(input *services.CommitStateDiffInput) (*servic
 		return &services.CommitStateDiffOutput{NextDesiredBlockHeight: lastCommittedBlock + 1}, nil
 	}
 
+	// if updating state records fails downstream the merkle tree entries will not bother us
+	s.merkle.Update(merkle.RootId(committedBlock), input.ContractStateDiffs)
```

Comment:
a bit confusing to all the variable `committedBlock` and not `committedBlockHeight`.. took me a while to figure this out